WordPress/WordPressTest/AccountSettingsRemoteInterfaceStub.swift

```diff
@@ -0,0 +1,77 @@
+@testable import WordPress
+import WordPressKit
+
+class AccountSettingsRemoteInterfaceStub: AccountSettingsRemoteInterface {
+
+    let updateSettingResult: Result<(), Error>
+    let getSettingsResult: Result<AccountSettings, Error>
+    let changeUsernameShouldSucceed: Bool
+    let suggestUsernamesResult: [String]
+    let updatePasswordResult: Result<(), Error>
+    let closeAccountResult: Result<(), Error>
+
+    init(
+        updateSettingResult: Result<Void, Error> = .success(()),
+        // Defaulting to failure to avoid having to create AccountSettings here, because it required an NSManagedContext
+        getSettingsResult: Result<AccountSettings, Error> = .failure(TestError()),
```

Comment:
If the intention of creating `TestError` type is to have something that conforms to `Swift.Error`, then maybe we can create a `NSError` instance here instead?

Reply:
> If the intention of creating `TestError` type is to have something that conforms to `Swift.Error`,

Yes, that's the idea

> then maybe we can create a `NSError` instance here instead?

🤔 I see your point. My thought with `TestError` was to have something to evolve into a flexible utility to place generic errors in tests. 

I might play with having `TestError` (or `ErrorForTests`?) as some kind of wrapper around `NSError` or maybe take a different route and extend `Error` to generate the value instead.

My end goal is for people not to have to think about how to create custom errors for tests when the error content doesn't matter.